_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,217 @@
+---
+title: "Stay on topic: Using Kafka to trigger your OpenFaaS functions"
+description: Martin outlines how you can use Kafka and subscribe OpenFaaS function to a topic using kafka-connector plugin.
+date: 2019-03-14
+image: /images/google-form-response/dots.jpg
+categories:
+  - kafka
+  - tutorial
+  - examples
+author_staff_member: martin
+dark_background: true
+---
+
+In this post I will show you how to subscribe your OpenFaaS functions to Kafka topics when running your functions via Kubernetes.
+
+OpenFaaS is production ready Serverless framework, and often finds itself as a part of bigger eco-system of software where the disparate parts may communicate with each-other in different ways. The OpenFaaS gateway exposes REST endpoints, but what if our services don't use HTTP?
+
+Kafka is stream-processing platform and at a high level the important concepts are Producers, Consumers and Brokers which communicate with each-other via massages attached to Topics. The way it works is that the Producer is sending messages on specific Topics to the Broker, then the Broker notifies all the Consumers which are interested in that Topic. The approach is fitting for distributed systems, because it decouples the communication between your services. 
+
+## Kafka Connector
+
+Kafka Connector plugin is the solution to your problem when it comes to integrating OpenFaaS as a component of a bigger distributed system which is using Kafka to make communication between its parts. You just need to apply annotation with the topic you would like to subscribe your function to and then run the connector with your Kafka configuration.
+
+Here is a diagram which shows how the component works:
+
+![Form Options](/images/kafka-connector/overview.jpg)
+
+## Pre-requisites
+
+First and most important thing is [running OpenFaaS instance](https://docs.openfaas.com/deployment/kubernetes/) with default configuration on Kubernetes, for our case [minikube deployment](https://kubernetes.io/docs/setup/minikube/) is more than enough.
+
+The [faas-cli](https://docs.openfaas.com/cli/install/) binary along with populated `OPENFAAS_URL` environmental variable pointing at your gateway.
+
+## Clone and configure Kafka Connector
+
+Pull the `kafka-connector` code from github:
+
+```bash
+$ git clone https://github.com/openfaas-incubator/kafka-connector.git
+```
+
+Inside `kafka-connector` folder navigate to `yaml/kubernetes`:
+
+```bash
+$ cd kafka-connector/yaml/kubernetes
+```
+
+The file `connector-dep.yaml` contains all the configuration you need to set for your connector.
```

Comment:
With helm this gets a lot simpler.

The topics section would be passed via a helm chart option.

The Kafka itself would be deployed from a Kafka chart and not from our chart.

Reply:
I think we discussed somewhere that the Kafka stack provided by wurstmeister does not have Charts repository. I will add installation through Charts for the Connector though

Reply:
Also can't seem to add list to key using Helm's `--set` flag. The output `... --set-string topics=topic1,topic2` gives me 
```
Error: failed parsing --set-string data: key "whatever2" has no value
```

Reply:
This part is done 👍